src/main/java/net/imglib2/algorithm/localextrema/LocalExtrema.java

```diff
@@ -92,79 +100,370 @@
+	 *            {@link LocalNeighborhoodCheck} that avoids the center pixel if
+	 *            <code>shape</code> does not skip the center pixel.
+	 * @return {@link ArrayList} of extrema
+	 */
+	public static < P, T > ArrayList< P > findLocalExtrema(
+			final RandomAccessible< T > source,
+			final Interval interval,
+			final LocalNeighborhoodCheck< P, T > localNeighborhoodCheck )
+	{
+		return findLocalExtrema( source, interval, localNeighborhoodCheck, new RectangleShape( 1, true ) );
+	}
+
+	/**
+	 * Find pixels that are extrema in their local neighborhood. The specific
+	 * test for being an extremum can be specified as an implementation of the
+	 * {@link LocalNeighborhoodCheck} interface.
+	 *
+	 * @param source
+	 *            Find local extrema within this {@link RandomAccessible}
+	 * @param interval
+	 *            Specifies the domain within which to look for extrema
+	 * @param localNeighborhoodCheck
+	 *            Check if current pixel qualifies as local maximum. It is the
+	 *            callers responsibility to pass a
+	 *            {@link LocalNeighborhoodCheck} that avoids the center pixel if
+	 *            <code>shape</code> does not skip the center pixel.
+	 * @param shape
+	 *            Defines the local neighborhood
+	 * @return {@link ArrayList} of extrema
+	 */
+	public static < P, T > ArrayList< P > findLocalExtrema(
+			final RandomAccessible< T > source,
+			final Interval interval,
+			final LocalNeighborhoodCheck< P, T > localNeighborhoodCheck,
+			final Shape shape )
+	{
+
+		final IntervalView< T > sourceInterval = Views.interval( source, interval );
+
+		final ArrayList< P > extrema = new ArrayList<>();
+
+		final Cursor< T > center = Views.flatIterable( sourceInterval ).cursor();
+		for ( final Neighborhood< T > neighborhood : shape.neighborhoods( sourceInterval ) )
+		{
+			center.fwd();
+			final P p = localNeighborhoodCheck.check( center, neighborhood );
+			if ( p != null )
+				extrema.add( p );
+		}
+
+		return extrema;
+
+	}
+
+	/**
+	 *
+	 * Get the required border size based on the bounding box of the
+	 * neighborhood specified by <code>shape</code>. This is useful for
+	 * determining by how much a {@link RandomAccessibleInterval} should be
+	 * expanded to include min and max positions in the local extrema search.
+	 *
+	 * @param shape
+	 *            Defines the local neighborhood
+	 * @param nDim
+	 *            Number of dimensions.
+	 * @return The required border size for the local neighborhood specified by
+	 *         <code>shape</code>
+	 */
+	public static long[] getRequiredBorderSize( final Shape shape, final int nDim )
+	{
+		final RandomAccessible< Neighborhood< Object > > neighborhood = shape.neighborhoodsRandomAccessible( ConstantUtils.constantRandomAccessible( new Object(), nDim ) );
+		final long[] min = LongStream.generate( () -> Long.MAX_VALUE ).limit( nDim ).toArray();
+		final long[] max = LongStream.generate( () -> Long.MIN_VALUE ).limit( nDim ).toArray();
```

Comment:
Because Java still lacks a function that generates an array that's filled with a constant value in a single line, something like `Arrays.create( int length, long value ) would be great. This is my little workaround for that. ;) At least for me, readability is improved wthat way.